### PR TITLE
Metrics for evaluation and coverage

### PR DESCRIPTION
**Pull request by bauersimon**

### CODE REVIEW

File: evaluate/metrics.go [hunk 1]
```diff
@@ -16,31 +17,37 @@ type Metrics struct {
 	Total uint
 	// Executed is the number of benchmarking candidates with successful execution.
 	Executed uint
+
+	// Coverage holds the coverage of the benchmarking candidates.
+	Coverage []float64
```
Comment (zimmski) [inline evaluate/metrics.go:22]: Do not use float, because it has problems to keep numbers straight. Use uint for metrics.

Comment (bauersimon) [reply evaluate/metrics.go:23]: But coverage reportings from Go have floating point digits. Or should we just cut them off?

**Comment by zimmski** [comment]
Will fix the test while i review

File: model/llm/llm.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ func NewLLMModel(provider provider.QueryProvider, modelIdentifier string) model.
 
 // llmGenerateTestForFilePrompt is the prompt used to query LLMs for test generation.
 var llmGenerateTestForFilePrompt = bytesutil.StringTrimIndentations(`
-	Given the following Go code file, provide a test file for this code.
+	Given the following Go code file "plain.test" with package "native", provide a test file for this code.
```
Comment (zimmski) [inline model/llm/llm.go:36]: Commit subject: "TMP"? Temporary? I usually prefix with TODO to make clear it is a TODO commit.

File: cmd/eval-symflower-codegen-testing/cmd/evaluate.go [hunk 3]
```diff
@@ -29,6 +29,8 @@ type Evaluate struct {
 	// TestdataPath determines the testdata path where all repositories reside grouped by languages.
 	TestdataPath string `long:"testdata" description:"Path to the testdata directory where all repositories reside grouped by languages." default:"testdata/"`
 
+	// CSVPath holds the path to the CSV results file.
+	CSVPath string `long:"csv" description:"Path to the CSV results file." default:"evaluation.csv"`
```
Comment (zimmski) [inline cmd/eval-symflower-codegen-testing/cmd/evaluate.go:33]: Let's group model-configurations and evaluation-configurations separately

Comment (zimmski) [reply cmd/eval-symflower-codegen-testing/cmd/evaluate.go:33]: Also, not that import that it is a CSV file for the field name.

File: cmd/eval-symflower-codegen-testing/cmd/evaluate.go [hunk 4]
```diff
@@ -112,6 +114,7 @@ func (command *Evaluate) Execute(args []string) (err error) {
 	// Evaluating models and languages.
 	log.Printf("Evaluating models and languages")
 	problemsPerModel := map[string][]error{}
+	metricsPerModel := map[string]evaluate.Metrics{}
```
Comment (zimmski) [inline cmd/eval-symflower-codegen-testing/cmd/evaluate.go:117]: Always sort.

File: evaluate/metrics.go (hunk 1 above)
Comment (zimmski) [reply evaluate/metrics.go:23]: Lets move it to the TODO. i didn't expect that we are collecting percentage. We should collect the absolute metrics and then do percentage as a summary. Not as a result of the benchmark. That is imprecise the more data we have.

**Pull request merged**